core/embed/rust/src/micropython/obj.rs

```diff
@@ -270,6 +280,17 @@ impl TryFrom<&str> for Obj {
     }
 }
 
+impl<const N: usize> TryFrom<String<N>> for Obj {
+    type Error = Error;
+
+    fn try_from(val: String<N>) -> Result<Self, Self::Error> {
+        // SAFETY:
+        //  - `str` is guaranteed to be UTF-8.
+        // EXCEPTION: Will raise if allocation fails.
+        catch_exception(|| unsafe { ffi::mp_obj_new_str(val.as_ptr().cast(), val.len()) })
```

Comment:
```suggestion
        (*val).try_into()
```
to call the `&str` impl